app/services/fees/fee-calculator.go

```diff
@@ -17,28 +17,110 @@
 package fees
 
 import (
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/provider"
 	"github.com/limechain/hedera-eth-bridge-validator/app/helper"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/shopspring/decimal"
+	"math"
 	"math/big"
 )
 
-func getFee() (*big.Int, error) {
-	return new(big.Int), nil
+type FeeCalculator struct {
+	rateProvider  provider.ExchangeRateProvider
+	configuration config.Hedera
 }
 
-func ValidateExecutionFee(strTransferFee string) (bool, error) {
-	transferFee, err := helper.ToBigInt(strTransferFee)
+func NewFeeCalculator(rateProvider provider.ExchangeRateProvider, configuration config.Hedera) *FeeCalculator {
+	return &FeeCalculator{
+		rateProvider:  rateProvider,
+		configuration: configuration,
+	}
+}
+
+func (fc FeeCalculator) ValidateExecutionFee(strTransferFee string, transferAmount string, gasPrice string) (bool, error) {
+	bigTransferAmount, err := helper.ToBigInt(transferAmount)
 	if err != nil {
-		return false, err
+		return false, InvalidTransferAmount
 	}
 
-	estimatedFee, err := getFee()
+	serviceFeePercent := new(big.Int).SetUint64(fc.configuration.Client.ServiceFeePercent)
+	bigServiceFee := new(big.Int).Mul(bigTransferAmount, serviceFeePercent)
+	bigServiceFee = new(big.Int).Div(bigServiceFee, new(big.Int).SetInt64(100))
```

Comment:
We should be charging service fee only on the transferred amount. At the moment we are charging a service fee for the `txReimbursementFee` as-well

We should do `(transferAmount - transferFee)*serviceFee/100` and not `transferAmount*serviceFee/100`